src/BuildScriptGenerator/DotNetCore/DotNetCoreBashBuildSnippet.sh.tpl

```diff
@@ -1,25 +1,182 @@
+SOURCE_DIR=$(pwd -P)
+
+if [ -z "$DESTINATION_DIR" ]
 then
-	outputDir="{{ PublishDirectory }}"
-	if [ -d "$outputDir" ]
+    DESTINATION_DIR="$SOURCE_DIR"
+fi
+
+if [ -d "$DESTINATION_DIR" ]
+then
+    cd $DESTINATION_DIR
+    DESTINATION_DIR=$(pwd -P)
+fi
+
+if [ ! -z "$INTERMEDIATE_DIR" ]
+then
+	echo "Using intermediate directory '$INTERMEDIATE_DIR'."
+	if [ ! -d "$INTERMEDIATE_DIR" ]
 	then
-		echo "Destination directory '$outputDir' already exists. Deleting it ..."
-		rm -rf "$outputDir"
+		echo
+		echo "Intermediate directory doesn't exist, creating it..."
+		mkdir -p "$INTERMEDIATE_DIR"		
 	fi
-else
-	outputDir="$DESTINATION_DIR"
+
+	cd "$INTERMEDIATE_DIR"
+	INTERMEDIATE_DIR=$(pwd -P)
+	cd "$SOURCE_DIR"
+	echo
+	echo "Copying files to the intermediate directory..."
+	START_TIME=$SECONDS
+	excludedDirectories=""
+	{{ for excludedDir in DirectoriesToExcludeFromCopyToIntermediateDir }}
+	excludedDirectories+=" --exclude {{ excludedDir }}"
+	{{ end }}
+	rsync --delete -rt $excludedDirectories . "$INTERMEDIATE_DIR"
+	ELAPSED_TIME=$(($SECONDS - $START_TIME))
+	echo "Done in $ELAPSED_TIME sec(s)."
+	SOURCE_DIR="$INTERMEDIATE_DIR"
+fi
+
+if [ "$SOURCE_DIR" == "$DESTINATION_DIR" ]
+then
+	DESTINATION_DIR="{{ PublishDirectory }}"
 fi
 
+echo
+echo "Source directory     : $SOURCE_DIR"
+echo "Destination directory: $DESTINATION_DIR"
+echo
+
+tmpDestinationDir="/tmp/puboutput"
+tmpDestinationPublishDir="$tmpDestinationDir/publish"
+zippedOutputFileName=oryx_output.tar.gz
+{{ if ZipAllOutput }}
+	ORIGINAL_DESTINATION_DIR="$DESTINATION_DIR"
+	DESTINATION_DIR="$tmpDestinationPublishDir"
+
+	# Clear the directory here so that pre-build script sees that change too
+	clearDirectory "$tmpDestinationDir"
+{{ end }}
+
+# Export the variables so that pre and post build scripts can access them
+export SOURCE_DIR
+export DESTINATION_DIR
+
+{{ if BenvArgs | IsNotBlank }}
+if [ -f /usr/local/bin/benv ]; then
+	source /usr/local/bin/benv {{ BenvArgs }}
+fi
+{{ end }}
+
+# Make sure to create the destination dir so that pre-build script has access to it
+mkdir -p "$DESTINATION_DIR"
+
+{{ if PreBuildScriptPath | IsNotBlank }}
+# Make sure to cd to the source directory so that the pre-build script runs from there
+cd "$SOURCE_DIR"
+"{{ PreBuildScriptPath }}"
+{{ end }}
+
 echo
 dotnetCoreVersion=$(dotnet --version)
 echo ".NET Core Version: $dotnetCoreVersion"
 
+cd "$SOURCE_DIR"
+
 echo
 echo Restoring packages ...
 echo
 dotnet restore "{{ ProjectFile }}"
 
+{{ if ZipAllOutput }}
+	publishToDirectory "$tmpDestinationPublishDir"
+
+	{{ if PostBuildScriptPath | IsNotBlank }}
+	# Make sure to cd to the source directory so that the post-build script runs from there
+	cd "$SOURCE_DIR"
+	"{{ PostBuildScriptPath }}"
+	{{ end }}
+
+	# Zip only the contents and not the parent directory
+	mkdir -p "$ORIGINAL_DESTINATION_DIR"
+	cd "$tmpDestinationPublishDir"
+	tar -zcf ../$zippedOutputFileName .
+	cd "$tmpDestinationDir"
+
+	if [ "$CLEAN_DESTINATION_DIR" == "true" ]
```

Comment:
is this PR still valid? Are you planning to update it with the changes from your other PR?